crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -158,11 +158,81 @@ impl Device {
     /// encrypted `m.room_key` event. This method determines if this `Device`
     /// can be confirmed as the creator and owner of the `m.room_key`.
     pub fn is_owner_of_session(&self, session: &InboundGroupSession) -> bool {
+        // An imported room key means that we did not receive the room key as a
+        // `m.room_key` event when the room key was initially exchanged.
+        //
+        // This could mean a couple of things:
+        //      1. We received the room key as a `m.forwarded_room_key`
+        //      2. We imported the room key through a file export.
+        //      3. We imoprted the room key through a backup.
+        //
+        // To be certain that a `Device` is the owner of a room key we need to have a
+        // proof that the `Curve25519` key of this `Device` was used to
+        // initially exchange the room key. This proof is provided by the Olm decryption
+        // step, see bellow for further clarification.
+        //
+        // Each of the above room key methods that receive room keys do not contain this
+        // proof and we received only a claim that the room key is tied to a
+        // `Curve25519` key.
+        //
+        // Since there's no way to verify that the claim is true, we say that we don't
+        // know that the room key belongs to this device.
```

Comment:
shouldn't this be inside the `if` block?

Reply:
Moved it.